api/apps/api/src/modules/projects/output-project-summaries/output-project-summaries.service.ts

```diff
@@ -0,0 +1,291 @@
+    @InjectRepository(OutputProjectSummaryApiEntity)
+    private readonly outputSummaryRepo: Repository<OutputProjectSummaryApiEntity>,
+  ) {}
+
+  async saveSummaryForProjectOfScenario(scenarioId: string): Promise<void> {
+    const { projectId } = await this.scenarioRepo.findOneOrFail({
+      where: { id: scenarioId },
+    });
+
+    const planningUnits = (await this.projectsPU.find({ where: { projectId } }))
+      .map((pu) => pu.puid)
+      .sort();
+
+    const scenarios = await this.scenarioRepo.find({
+      select: { id: true, projectScenarioId: true, name: true },
+      where: { projectId },
+    });
+
+    const bestSolutionData = await this.getBestSolutionDataForScenarios(
+      scenarios,
+    );
+    const summedSolutionData = await this.getSummedSolutionDataForScenarios(
+      scenarios,
+    );
+
+    const workingDirectory = OutputProjectSummariesService.getProjectTempFolder(
+      projectId,
+    );
+    const csvPath = joinPath(workingDirectory, outputProjectSummaryFolder);
+    await this.createCSVFolder(csvPath);
+
+    const summaryCSVStream = this.arrangeSummaryCSV(
+      planningUnits,
+      scenarios,
+      bestSolutionData,
+      summedSolutionData,
+    );
+    const scenarioCSVStream = this.arrangeScenariosCSV(scenarios);
+
+    await this.writeCSV(
+      workingDirectory,
+      outputProjectSummaryFilename,
+      summaryCSVStream,
+    );
+    await this.writeCSV(
+      workingDirectory,
+      outputProjectSummaryScenariosFilename,
+      scenarioCSVStream,
+    );
+
+    const zipFullPath = this.zipCSVs(projectId, workingDirectory);
+
+    await this.saveSummary(projectId, zipFullPath);
+
+    await this.cleanupProjectTempFolder(projectId);
+  }
+
+  private arrangeSummaryCSV(
+    planningUnits: number[],
+    scenarios: Scenario[],
+    bestSolutionData: ProjectScenarioBestSolutionMapping,
+    summedSolutionData: ProjectScenarioSummedSolutionMapping,
+  ): CsvFormatterStream<FormatterRow, FormatterRow> {
+    const projectScenarioIds = scenarios
+      .map((scenario) => scenario.projectScenarioId)
+      .sort();
+
+    // The headers of the CSV file is dependent on the set of scenarios for the project
+    // it needs to be preocomputed in ascending order
+    // The CSV rows will also be sorted in ascending order by puid
+    // Headers will follow this format: puid, Snnn_best, Snnn_ssoln, Snnn_best, Snnn_ssoln....
+    // a set of 2 column per each scenario, with nnn being the projectScenarioId (within the project)
+    // keep in mind that the projectScenarioId might have gaps in the sequence
+    const headers = [];
+    headers.push('puid');
+    for (const projectScenarioId of projectScenarioIds) {
+      const scenarioId = this.formatProjectScenarioId(projectScenarioId);
+      headers.push(`S${scenarioId}_best`);
+      headers.push(`S${scenarioId}_ssoln`);
+    }
+
+    const csvData = [];
+    csvData.push(headers);
+    for (const planningUnit of planningUnits.sort()) {
```

Comment:
same as above - we don't want a lexicographical sorting but a numeric sorting here